Remove Python 3.6 from PR checks

JiriPapousek opened the pull request:
# Description

Python 3.6 is at its EOF now. Its support would prevent us from using functionalities from later versions.

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps

NA

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

tisnik reviewed (commented):
- why the coverage hack was removed?
- what about recent Python versions?

JiriPapousek commented:
> * why the coverage hack was removed?
> * what about recent Python versions?

I added the coverage back (now I understood what it is for). Added PR check for Python 3.11.

tisnik reviewed (approved):
ok
